File: csharp/collatz-conjecture/CollatzConjecture.cs
using System;

public static class CollatzConjecture
{
    public static int Steps(int number)
    {
        if (number <= 0)
        {
            throw new ArgumentOutOfRangeException("Number must be positive");
        }

        int steps = 0;
        while (number != 1)
        {
            number = number % 2 == 0 ? number / 2 : number * 3 + 1;
            steps++;
        }

        return steps;
    }
}

File: typescript/collatz-conjecture/collatz-conjecture.ts
export function steps(count: number): number {
  let steps = 0
    if (count <= 0 || !Number.isInteger(count)) {
        throw new Error('Only positive integers are allowed')
    }
    while (count !== 1) {
        if (count % 2 === 0) {
            count = count / 2
        } else {
            count = count * 3 + 1
        }
        steps++
    }
    return steps
}
